homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,282 @@
+        )
+
+        parametersDict = {
+            parameter.parameter_code: parameter for parameter in parameters
+        }
+        self.device_id = device.id
+        self.available_presets: Parameter | None = parametersDict.get(PARAM_PRESET_MODE)
+        self.available_fan_modes: Parameter | None = parametersDict.get(PARAM_FAN_MODE)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        value = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_CURRENT_TEMPERATURE
+        )
+        if value is None:
+            return None
+        return float(value.value)
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        value = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_TARGET_TEMPERATURE
+        )
+        if value is None:
+            return None
+        return float(value.value)
+
+    @cached_property
+    def preset_modes(self) -> list[str] | None:
+        """Return the available preset modes."""
+        if self.available_presets is None or self.available_presets.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_presets.details
+            if item is not None
+        ]
+
+    @cached_property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_fan_modes.details
+            if item is not None
+        ]
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        preset_mode = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_PRESET_MODE
+        )
+
+        if preset_mode is None:
+            return None
+        if self.available_presets is None or self.available_presets.details is None:
+            _LOGGER.debug("available_presets is None")
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_presets.details
+                if item is not None and item.state == preset_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            return None
+        return val.description
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        fan_mode = self.coordinator.data[self.device_id].state.get_parameter_value(
+            PARAM_FAN_MODE
+        )
+        if fan_mode is None:
+            return None
+
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            _LOGGER.debug("available_fan_modes is None")
```

Comment:
is this an expected/typical situation? if not, this should be an error or warning log, not debug.

Reply:
same for the other debug logs in this file